spotbugs/etc/messages.xml

```diff
@@ -9128,6 +9137,51 @@ Using floating-point variables should not be used as loop counters, as they are
       ]]>
         </Details>
   </BugPattern>
+
+  <BugPattern type="ENV_USE_PROPERTY_INSTEAD_OF_ENV">
+    <ShortDescription>It is preferable to use portable Java property instead of environment variable.</ShortDescription>
+    <LongDescription>It is preferable to use portable Java property '{3}' instead of environment variable '{2}' in method {1}.</LongDescription>
+    <Details>
+      <![CDATA[<p>
+        Environment variables are not portable, the variable name itself (not only the value) may be different depending on the running OS.
+        Not only the names of the specific environment variables can differ (e.g. `USERNAME` in Windows and `USER` in Unix systems),
+        but even the semantics differ, e.g. the case sensitivity (Windows being case-insensitive and Unix case-sensitive).
+        Thus, the environment variables may not obey the general contract of the <code>Object.equals(java.lang.Object)</code> and <code>Object.hashCode()</code> methods.
+        Consequently, environment variables may have unintended side effects.
+        Also, The visibility of environment variables are also less restricted compared to Java Properties: they are visible to all descendants
+        of the defining process, not just the immediate Java subprocess. Environment variables may be accidentally or maliciously altered during
```

Comment:
> Environment variables may be accidentally or maliciously altered during
        process execution using the <code>java.lang.ProcessBuilder</code> class.

Another piece of wrong claims. As I've mentioned before, if attacker or user can control command line creation via ProcessBuilder, he can change **everything**, not only variables. Please remove this false sentence.

Reply:
Thank you for clarifying this part. You are right, if someone has access to the ProcessBuilder or some similar construction, then everything can be changed.
The sentence itself is not entirely wrong, since environment variables can be changed, but it is misleading, indicating that the java properties can't be, so I removed it.